Coverage CI failing on icu_datetime

The errors look like:

```
Error: test failed, to rerun pass '-p icu_datetime --lib'
Caused by:
  process didn't exit successfully: `/home/runner/work/icu4x/icu4x/target/debug/deps/icu_datetime-4fefaf1a9480d177` (signal: 6, SIGABRT: process abort signal)
Error: The process '/usr/share/rust/.cargo/bin/cargo' failed with exit code 101
```

I can't see any other info in the log to hint at the source of the error.  The error first occured on main after the following commit:

https://github.com/unicode-org/icu4x/commit/2d6664a376df209f86d4af47f0a686b91b8172e0

I tried to debug it today unsuccessfully.
I tried to reproduce it locally via https://github.com/mozilla/grcov#example-how-to-generate-source-based-coverage-for-a-rust-project and ended up with:

```
test src/subtags/variants.rs - subtags::variants::Variants::into_raw (line 83) ... ok

test result: ok. 93 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.49s

mozilla@mozilla:~/projects/icu4x$ grcov . -s . --binary-path ./target/debug/ -t html --branch --ignore-not-existing -o ./target/debug/coverage/
00:16:53 [ERROR] Error while executing llvm tools: Failed to execute "/home/mozilla/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-profdata" "merge" "-sparse" "/tmp/.tmpQBBZAK/utils/litemap/default_1.profraw" "/tmp/.tmpQBBZAK/resources/testdata/default_1.profraw" "/tmp/.tmpQBBZAK/experimental/bies/default_1.profraw" "/tmp/.tmpQBBZAK/tools/benchmark/macros/default_1.profraw" "/tmp/.tmpQBBZAK/components/ecma402/default_1.profraw" "/tmp/.tmpQBBZAK/tools/benchmark/memory/default_1.profraw" "/tmp/.tmpQBBZAK/components/plurals/default_1.profraw" "/tmp/.tmpQBBZAK/components/uniset/default_1.profraw" "/tmp/.tmpQBBZAK/default_1.profraw" "/tmp/.tmpQBBZAK/utils/writeable/default_1.profraw" "/tmp/.tmpQBBZAK/components/icu4x/default_1.profraw" "/tmp/.tmpQBBZAK/experimental/segmenter_lstm/default_1.profraw" "/tmp/.tmpQBBZAK/utils/fixed_decimal/default_1.profraw" "/tmp/.tmpQBBZAK/components/provider_cldr/default_1.profraw" "/tmp/.tmpQBBZAK/components/locid/default_1.profraw" "/tmp/.tmpQBBZAK/components/provider_fs/default_1.profraw" "/tmp/.tmpQBBZAK/components/locid/macros/default_1.profraw" "/tmp/.tmpQBBZAK/components/icu/default_1.profraw" "/tmp/.tmpQBBZAK/components/datetime/default_1.profraw" "/tmp/.tmpQBBZAK/components/provider/default_1.profraw" "/tmp/.tmpQBBZAK/components/locale_canonicalizer/default_1.profraw" "/tmp/.tmpQBBZAK/components/provider_ppucd/default_1.profraw" "-o" "/tmp/.tmpQBBZAK/1/grcov.profdata"
No such file or directory (os error 2)
```

Anyone has an idea on how to complete grcov using source-based coverage locally?

--- a/.github/workflows/build-test.yml
+++ b/.github/workflows/build-test.yml
@@ -293,12 +293,12 @@ jobs:
       # then this line is no longer needed.
       - uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly-2021-03-08
+          toolchain: nightly-2021-03-15
           override: true
 
       - name: Run the example with dhat-rs to collect memory information
         run: |
-          cargo run --package icu_benchmark_memory -- --os ${{ matrix.os }} ${{ matrix.examples }} --toolchain nightly-2021-03-08
+          cargo run --package icu_benchmark_memory -- --os ${{ matrix.os }} ${{ matrix.examples }} --toolchain nightly-2021-03-15
 
       # Benchmarking & dashboards job > (unmerged PR only) Convert benchmark output into
       # dashboard HTML in a commit of a branch of the local repo.

--- a/.github/workflows/coverage.yml
+++ b/.github/workflows/coverage.yml
@@ -24,7 +24,7 @@ jobs:
 
       - uses:                   actions-rs/toolchain@v1
         with:
-          toolchain:            nightly-2021-03-08
+          toolchain:            nightly-2021-03-01
           override:             true
 
       - uses:                   actions-rs/cargo@v1

--- a/.github/workflows/coverage.yml
+++ b/.github/workflows/coverage.yml
@@ -24,7 +24,7 @@ jobs:
 
       - uses:                   actions-rs/toolchain@v1
         with:
-          toolchain:            nightly-2021-03-01
+          toolchain:            nightly-2021-03-15
           override:             true
 
       - uses:                   actions-rs/cargo@v1

